portfolio/src/main/webapp/script.js

```diff
@@ -12,6 +12,91 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+// Loads the comments on every page without having to redundantly specify in HTML.
+window.onload = function() {
+  loadCommonFeatures();
+  // Only load map on pages where the "map" div is present.
+  if (document.getElementById('map') !== null) {
+    loadMap();
+  }
+}
+
+/** Loads HTML features which are common to all pages of the website */
+// TODO: Possibly find a way to do this which doesn't rebuild all of these
```

Comment:
You could explore storing the original HTML in a "partial.html" file and then using AJAX to fetch that html content and insert it.

Alternatively, you could write all of the html in the ".innerHTML" string and the brower will parse it.